src/odemis/acq/stream/_sync.py

```diff
@@ -1048,38 +1062,39 @@ def _stopLeeches(self):
         for l in self.leeches:
             l.complete(self.raw)
 
-    def _waitForImage(self, start, rep_time):
+    def _waitForImage(self, det_num, px_time):
         """
         Wait for the detector to acquire the image
-        :param start (0<float): start time for acquisition
-        :param rep_time (0<float): dwell time of one pixel
+        :param px_time (0<float): estimated time spend for one pixel
         :return (bool): True if acquisition timed out
         """
 
         # A big timeout in the wait can cause up to 50 ms latency.
         # => after waiting the expected time only do small waits
 
-        # wait for detector to acquire image
-        endt = start + rep_time * 3 + 5
-        timedout = not self._acq_complete[-1].wait(rep_time + 0.01)
+        start = time.time()
+        endt = start + px_time * 3 + 5
+        timedout = not self._acq_complete[self._det_num].wait(px_time + 0.01)
         if timedout:
-            logging.debug("Waiting more for rep")
+            logging.debug("Waiting a bit more for detector %d to acquire image." % self._det_num)
             while time.time() < endt:
-                timedout = not self._acq_complete[-1].wait(0.005)
+                timedout = not self._acq_complete[self._det_num].wait(0.005)
                 if not timedout:
                     break
-        logging.debug("Got rep synchronisation")
+        logging.debug("Got synchronized acquisition from detector %d." % self._det_num)
 
         return timedout
 
-    def _acquireImage(self, n, px_idx, rep_time, sem_time, sub_pxs,
+    def _acquireImage(self, det_num, n, px_idx, px_time, sem_time, sub_pxs,
```

Comment:
No, let's not use `det_num` as an argument here. Because this function should know explicitly that it wants to acquire all the streams, and that the ccd ID is in `self._ccd_num`. (Actually, it seems the argument is never used, which is good).